incident.go

```diff
@@ -653,4 +653,57 @@ func (c *Client) ManageIncidentAlertsWithContext(ctx context.Context, incidentID
 	return &result, nil
 }
 
-/* TODO: Create Status Updates */
+// IncidentStatusUpdate is a status update for the specified incident.
+type IncidentStatusUpdate struct {
+	ID        string    `json:"id,omitempty"`
+	Message   string    `json:"message,omitempty"`
+	CreatedAt string    `json:"created_at,omitempty"`
+	Sender    APIObject `json:"sender,omitempty"`
+}
+
+// CreateIncidentStatusUpdateResponse is returned from the API as a response to creating an incident status update.
+type CreateIncidentStatusUpdateResponse struct {
+	IncidentStatusUpdate IncidentStatusUpdate `json:"status_update"`
+}
+
+// CreateIncidentStatusUpdateWithContext creates a new status update for the specified incident.
+func (c *Client) CreateIncidentStatusUpdateWithContext(ctx context.Context, id string, statusUpdate IncidentStatusUpdate) (*IncidentStatusUpdate, error) {
```

Comment:
I think there are a few things we should change with this method signature before merging.

* It should be renamed to `CreateIncidentStatusUpdate`.
  * The methods that are named `...WithContext` were added so we could introduce `context.Context` support without breaking API compatibility. New methods should accept the `context.Context` and not use this naming scheme.
* Instead of accepting a `statusUpdate IncidentStatusUpdate`, I'd rather see this take a `from, message string`.
  * This will make it work more like other methods that require a `from` parameter, and it more closely matches the [REST API](https://developer.pagerduty.com/api-reference/b3A6Mjc0ODE1Mw-create-a-status-update-on-an-incident) which only needs the string value of the `message`, and not the full `IncidentStatusUpdate` JSON object.
* This should not return a pointer (`*IncidentStatusUpdate`) and instead return a concrete `IncidentStatusUpdate` value.

Reply:
Ah. OK. You mean, 
```
CreateIncidentStatusUpdate(ctx context.Context, id string, from string, message string)
```

This is the best. Right?

Reply:
Yeah, that looks good to me. The complete header would end up being:

```Go
func (c *Client) CreateIncidentStatusUpdateWithContext(ctx context.Context, id string, from string, message string) (*IncidentStatusUpdate, error)
```

Reply:
Yeah, that looks good to me. The complete header would end up being:

```Go
func (c *Client) CreateIncidentStatusUpdate(ctx context.Context, id string, from string, message string) (*IncidentStatusUpdate, error)
```

Reply:
Yeah, that looks good to me. The complete header would end up being:

```Go
func (c *Client) CreateIncidentStatusUpdate(ctx context.Context, id string, from string, message string) (IncidentStatusUpdate, error)
```

Reply:
>This should not return a pointer (*IncidentStatusUpdate) and instead return a concrete IncidentStatusUpdate value.

Sorry, why do you think that only this thread should not return a pointer?

Reply:
The primary purpose of pointers are to permit shared mutability of some value, with a secondary purpose to be to differentiate between there being no value and a zero value (think `*string` when using `omitempty` with JSON).

In this case, were are returning a result from the API. The client isn't going to update the value at all once it's returned, and so there isn't a need for shared mutability. The value probably won't be mutated at all, and just read.

To my second purpose, we don't need to differentiate between no value and the zero value, because us returning an `error` handles that. If the `error` is not nil, the other returned values should be discarded.

Reply:
Thank you for your reply. I agree with your point.

But I mean, other similar method (e.g. CreateIncidentNotes etc) return a pointer. If only this method return value, it might be confused for developers, I thought. 

Ok. let me change to use value instead of a pointer.

Reply:
Yeah. That is definitely true and I appreciate you calling that out.

For v2+ I plan to review all of those return types, and make sure we only return a pointer when it makes sense. I created an issue a few months back to track this: https://github.com/PagerDuty/go-pagerduty/issues/270.

So until v2 comes along, I'm leaning in to some inconsistency to reduce the number of breaking changes we will need to make in the future.

Thank you for changing this!